### PR TITLE
579 - ODS-Module - About initialization

### PR DESCRIPTION
**Pull request by paulinea**

### CODE REVIEW

File: module-about/src/main/java/com/orange/ods/module/about/OdsAboutFileScreen.kt [hunk 1]
```diff
@@ -17,29 +17,27 @@ import android.webkit.WebView
...
+import com.orange.ods.module.about.utilities.extension.injectLightDarkModeCss
+import com.orange.ods.module.about.utilities.extension.launchUrl
 import java.io.BufferedReader
 import java.nio.charset.StandardCharsets
 
 private const val FilePath = "file:///android_res/raw/"
 
 @Composable
-fun AboutFileScreen(aboutItemId: Long) {
+fun OdsAboutFileScreen(aboutItemId: Long, darkModeEnabled: Boolean) {
```
Comment (florentmaitre) [inline module-about/src/main/java/com/orange/ods/module/about/OdsAboutFileScreen.kt:36]:
```suggestion
internal fun OdsAboutFileScreen(aboutItemId: Long, darkModeEnabled: Boolean) {
```

File: module-about/src/main/res/drawable-xxxhdpi/il_about.png
Comment (florentmaitre) [inline module-about/src/main/res/drawable-xxxhdpi/il_about.png:1]: mdpi version is missing. I think we already discussed this point for other resources but it seems that mdpi is still used:

https://developer.android.com/about/dashboards/index.html (see `Screen sizes and densities`)

File: module-about/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  ~ * license that can be found in the LICENSE file or at
+  ~ * https://opensource.org/licenses/MIT.
+  ~ */
+  -->
+
+<resources>
+    <string name="about_top_bar_title">About</string>
+
+    <string name="app_share">Share</string>
+    <string name="app_feedback">Feedback</string>
```
Comment (florentmaitre) [inline module-about/src/main/res/values/strings.xml:15]: Should we prefix these names with `about`?

File: module-about/src/main/java/com/orange/ods/module/about/configuration/OdsAboutModuleConfiguration.kt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    /**
+     * The optional actions displayed in the overflow menu of the About module TopAppBar. If the list is empty, the overflow menu icon will not be displayed.
+     */
+    val topAppBarOverflowMenuActions: List<OdsTopAppBarOverflowMenuActionItem> = emptyList(),
+
+    ) {
+
+    override fun toString(): String {
+        return Uri.encode(Gson().toJson(this))
```
[the remark this answers is not in the document]
Comment (paulinea) [reply module-about/src/main/java/com/orange/ods/module/about/configuration/OdsAboutModuleConfiguration.kt:77]: This has been removes in about step 2

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -372,7 +372,23 @@
...
+    <string name="module_about_description">An « About » screen should be displayed in all Orange applications to show the application name, software version as well as all legal data protection, privacy, terms of service and accessibility statement.</string>
+    <string name="module_about_customization">Some items such as: illustration (customizable), app name, privacy policy, terms of service and accessibility statement are mandatory.\nTo this standard screen you can add optional components:</string>
+    <string name="module_about_customization_version">Version</string>
+    <string name="module_about_customization_description">Description</string>
+    <string name="module_about_customization_share">Share</string>
+    <string name="module_about_customization_feedback">Feedback</string>
+    <string name="module_about_demo_app_name">Application Name</string>
+    <string name="module_about_demo_version">Version</string>
+    <string name="module_about_demo_description">Add here a short description of the application.</string>
+    <string name="module_about_demo_feedback_button_element">feedback button</string>
```
[the remark this answers is not in the document]
Comment (paulinea) [reply app/src/main/res/values/strings.xml:389]: Actually, it is not the title of the button but the name of the element to display in the toast when we click on it.

File: module-about/src/main/java/com/orange/ods/module/about/configuration/OdsAboutItem.kt [hunk 5]
```diff
@@ -39,10 +38,4 @@ class UrlAboutItem(
...
     get() = aboutItems.indexOf(this).toLong()
 
-val aboutItems = listOf(
-    UrlAboutItem(R.string.about_menu_design_guidelines, "https://system.design.orange.com/0c1af118d/p/019ecc-android/"),
-    FileAboutItem(R.string.about_menu_legal_notice, R.raw.about_legal_notice, FileAboutItem.FileFormat.Html),
-    FileAboutItem(R.string.about_menu_privacy_policy, R.raw.about_privacy_policy, FileAboutItem.FileFormat.Html),
-    FileAboutItem(R.string.about_menu_changelog, R.raw.changelog, FileAboutItem.FileFormat.Markdown),
-    UrlAboutItem(R.string.about_menu_report_issue, "https://github.com/Orange-OpenSource/ods-android/issues/new/choose"),
-)
+val aboutItems = listOf<AboutItem>()
```
[the remark this answers is not in the document]
Comment (paulinea) [reply module-about/src/main/java/com/orange/ods/module/about/configuration/OdsAboutItem.kt:41]: Yes this is done in the other PR

File: module-about/src/main/java/com/orange/ods/module/about/navigation/OdsAboutModuleNavigation.kt [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import androidx.navigation.NavOptions
+import androidx.navigation.compose.composable
+import com.orange.ods.module.about.OdsAboutMainScreen
+import com.orange.ods.module.about.OdsAboutViewModel
+import com.orange.ods.module.about.configuration.LocalOdsAboutModuleConfiguration
+
+const val OdsAboutRoute = "ods/module/about/"
+const val OdsAboutDemoRoute = "ods/module/about/demo"
+
+fun NavController.navigateToOdsAboutDemo(navOptions: NavOptions? = null) {
```
[the remark this answers is not in the document]
Comment (paulinea) [reply module-about/src/main/java/com/orange/ods/module/about/navigation/OdsAboutModuleNavigation.kt:25]: I don't know how to do...

File: app/src/main/java/com/orange/ods/app/ui/modules/ModuleDetailColumn.kt [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    module: Module,
+    content: @Composable () -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .verticalScroll(rememberScrollState())
+            .padding(bottom = dimensionResource(id = com.orange.ods.R.dimen.screen_vertical_margin))
+    ) {
+        DetailScreenHeader(
+            imageRes = DrawableManager.getDrawableResIdForCurrentTheme(resId = module.imageRes),
```
[the remark this answers is not in the document]
Comment (paulinea) [reply app/src/main/java/com/orange/ods/app/ui/modules/ModuleDetailColumn.kt:36]: Actually the generic image is not the good one I think but I add this `imageContentScale` parameter to `DetailScreenHeader` cause same problem can happen.

File: module-about/src/main/java/com/orange/ods/module/about/configuration/OdsAboutModuleConfiguration.kt [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+     * If not provided, the default Orange illustration will be displayed.
+     */
+    @DrawableRes
+    val appIllustration: Int = R.drawable.il_about,
+
+    /**
+     * The application version displayed on the main screen of the About module.
+     * If null, no version will be displayed.
+     */
+    val appVersion: String? = null,
```
[the remark this answers is not in the document]
Comment (paulinea) [reply module-about/src/main/java/com/orange/ods/module/about/configuration/OdsAboutModuleConfiguration.kt:44]: As seen together, the about module provides now a `VersionHelper` which can generate a version String based on package information. I added the information in the kdoc.

File: app/src/main/res/values/strings.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (florentmaitre) [reply app/src/main/res/values/strings.xml:389]: My bad, I read this one too fast.

**Review by florentmaitre — APPROVED** [review]

**Pull request merged**